app/models/miq_queue.rb

```diff
@@ -278,12 +278,35 @@ def self.put_or_update(find_options)
   #   puts a new item on the queue.  If the item was found, it will not be
   #   changed, and will be yielded to an optional block, generally for logging
   #   purposes.
+
+  # Return task.id
   def self.put_unless_exists(find_options)
-    put_or_update(find_options) do |msg, item_hash|
-      ret = yield(msg, item_hash) if block_given?
-      # create the record if the original message did not exist, don't change otherwise
-      ret if msg.nil?
+    find_options = default_get_options(find_options)
+    conds = find_options.dup
+
+    task = MiqTask.find_by(:identifier => conds.hash.to_s)
+
+    save_options = block_given? ? yield(task, find_options) : nil
```

Comment:
I'm pretty sure we have updated all `put_unless_exists` to ignore any return values.
If at all possible, let's move the yield to after we create the queue entry.

removing `save_options` vs `put_options` buys us a lot.